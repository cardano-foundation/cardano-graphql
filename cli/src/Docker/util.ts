import { QuestionCollection } from 'inquirer'
import { DockerComposeStack, StatefulService } from './index'

export function selectServicesFromStack (stack: DockerComposeStack, multi = true): QuestionCollection {
  const label = multi === true ? 'services' : 'service'
  return {
    name: label,
    message: `Select ${label}`,
    type: multi === true ? 'checkbox' : 'list',
    choices: [...stack.statefulServices.values()].map(service => {
      return {
        name: service.name,
        value: service
      }
    }),
    validate (input) {
      if (input.length === 0) {
        throw new Error('At least one service must be selected')
      }
      return true
    }
  }
}

export function servicesFromNameArg (stack: DockerComposeStack, names: string): StatefulService[] {
  return names.split(',').map((name: string) => {
    if (!stack.statefulServices.has(name)) {
      throw new Error(`Unknown service ${name}`)
    }
    return stack.statefulServices.get(name)
  })
}
