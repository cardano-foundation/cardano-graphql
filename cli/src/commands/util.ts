import { DockerComposeStack } from '../docker'

export function serviceChoices (stack: DockerComposeStack) {
  return [...stack.statefulServices.values()].map(service => {
    return { name: service.name, value: service }
  })
}

export function serviceFromName (stack: DockerComposeStack, name: string) {
  if (!stack.statefulServices.has(name)) {
    throw new Error(`Unknown service ${name}`)
  }
  return stack.statefulServices.get(name)
}

export function servicesFromNames (stack: DockerComposeStack, names: string) {
  return names.split(',').map((name: string) => serviceFromName(stack, name))
}
