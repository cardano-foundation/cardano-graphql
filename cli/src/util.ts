
export function ensureValue (input: string) {
  return !input ? 'Cannot be blank' : true
}
