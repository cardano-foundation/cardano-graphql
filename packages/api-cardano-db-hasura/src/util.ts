import { ConnectionConfig, createInteractionContext } from '@cardano-ogmios/client'
import { Logger } from 'ts-log'

export const createInteractionContextWithLogger = (connection: ConnectionConfig, logger: Logger, module: string, onClose?: () => Promise<void>) =>
  createInteractionContext(
    (error) => {
      logger.error({ module, error }, error.message)
    },
    async (code, reason) => {
      if (code === 1000) {
        logger.info({ module, code }, reason)
      } else {
        logger.error({ module, code }, 'Connection closed')
        await onClose?.()
      }
    },
    {
      connection
    }
  )
