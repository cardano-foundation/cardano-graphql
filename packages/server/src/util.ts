import { Application } from 'express'
import http from 'http'

export function listenPromise (app: Application, port: number, hostname?: string): Promise<http.Server> {
  return new Promise(function (resolve, reject) {
    const server: http.Server = app.listen(port, hostname, () => resolve(server))
    server.on('error', reject)
  })
}
