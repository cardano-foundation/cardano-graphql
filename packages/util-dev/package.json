{
  "name": "@cardano-graphql/util-dev",
  "version": "1.0.0",
  "description": "Common development utilities",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -b ./src",
    "cleanup": "shx rm -rf dist node_modules",
    "lint": "eslint --ignore-path ../../.eslintignore \"**/*.ts\"",
    "prepack": "yarn build",
    "test": "shx echo No tests in this package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/input-output-hk/cardano-graphql.git"
  },
  "author": "Rhys Bartels-Waller",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/input-output-hk/cardano-graphql/issues"
  },
  "homepage": "https://github.com/input-output-hk/cardano-graphql/blob/master/packages/util-dev/README.md",
  "devDependencies": {
    "@cardano-graphql/util": "1.0.0",
    "shx": "^0.3.2"
  },
  "directories": {
    "lib": "src",
    "test": "test"
  },
  "files": [
    "dist",
    "LICENSE",
    "NOTICE",
    "README.md"
  ]
}
