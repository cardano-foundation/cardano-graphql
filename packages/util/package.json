{
  "name": "@cardano-graphql/util",
  "version": "1.0.0",
  "description": "Common utilities",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "scripts": {
    "build": "tsc -b ./src",
    "cleanup": "shx rm -rf dist node_modules",
    "lint": "eslint --ignore-path ../../.eslintignore \"**/*.ts\"",
    "prepack": "yarn build",
    "test": "shx echo No tests in this package"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/input-output-hk/cardano-graphql.git"
  },
  "author": "Rhys Bartels-Waller",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/input-output-hk/cardano-graphql/issues"
  },
  "homepage": "https://github.com/input-output-hk/cardano-graphql/blob/master/packages/util/README.md",
  "dependencies": {
    "graphql": "14.5.8",
    "graphql-bigint": "^1.0.0",
    "graphql-tag": "^2.10.3",
    "p-retry": "^4.2.0"
  },
  "devDependencies": {
    "@types/node": "^14.0.13",
    "shx": "^0.3.2",
    "typescript": "^3.9.5"
  },
  "directories": {
    "lib": "src",
    "test": "test"
  },
  "files": [
    "dist",
    "LICENSE",
    "NOTICE",
    "README.md"
  ]
}
