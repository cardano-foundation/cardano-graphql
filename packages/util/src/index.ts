import { onFailedAttemptFor } from './onFailedAttemptFor'
import { loadQueryNode } from './queryNodeLoading'
import * as dateTimeValidation from './dateTimeValidation'
import * as scalars from './scalars'

export default {
  dateTimeValidation,
  onFailedAttemptFor,
  loadQueryNode,
  scalars
}
