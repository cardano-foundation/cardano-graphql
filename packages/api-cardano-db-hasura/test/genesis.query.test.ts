import path from 'path'
import { DocumentNode } from 'graphql'
import util from '@cardano-graphql/util'
import { TestClient } from '@cardano-graphql/util-dev'
import { testClient } from './util'

function loadQueryNode (name: string): Promise<DocumentNode> {
  return util.loadQueryNode(path.resolve(__dirname, '..', 'src', 'example_queries', 'genesis'), name)
}

describe('genesis', () => {
  let client: TestClient

  beforeAll(async () => {
    client = await testClient.preprod()
  })

  it('Returns key information about the network genesis', async () => {
    const query = { query: await loadQueryNode('keyNetworkInfo') }
    const result = await client.query(query)
    expect(result.data).toMatchSnapshot()
  })
})
