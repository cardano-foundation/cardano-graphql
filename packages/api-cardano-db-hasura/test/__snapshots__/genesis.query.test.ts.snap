// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genesis Returns key information about the network genesis 1`] = `
Object {
  "genesis": Object {
    "alonzo": Object {
      "collateralPercentage": 150,
      "executionPrices": Object {
        "prMem": Object {
          "denominator": 10000,
          "numerator": 577,
        },
        "prSteps": Object {
          "denominator": 10000000,
          "numerator": 721,
        },
      },
      "lovelacePerUTxOWord": 34482,
      "maxBlockExUnits": Object {
        "exUnitsMem": 50000000,
        "exUnitsSteps": 40000000000,
      },
      "maxCollateralInputs": 3,
      "maxTxExUnits": Object {
        "exUnitsMem": 10000000,
        "exUnitsSteps": 10000000000,
      },
      "maxValueSize": 5000,
    },
    "byron": Object {
      "blockVersionData": Object {
        "maxBlockSize": 2000000,
        "maxHeaderSize": 2000000,
        "maxProposalSize": 700,
        "maxTxSize": 4096,
        "scriptVersion": 0,
        "slotDuration": 20000,
      },
      "protocolConsts": Object {
        "k": 2160,
        "protocolMagic": 1,
      },
      "startTime": 1654041600,
    },
    "shelley": Object {
      "activeSlotsCoeff": 0.05,
      "epochLength": 432000,
      "genDelegs": Object {
        "637f2e950b0fd8f8e3e811c5fbeb19e411e7a2bf37272b84b29c1a0b": Object {
          "delegate": "aae9293510344ddd636364c2673e34e03e79e3eefa8dbaa70e326f7d",
          "vrf": "227116365af2ed943f1a8b5e6557bfaa34996f1578eec667a5e2b361c51e4ce7",
        },
        "8a4b77c4f534f8b8cc6f269e5ebb7ba77fa63a476e50e05e66d7051c": Object {
          "delegate": "d15422b2e8b60e500a82a8f4ceaa98b04e55a0171d1125f6c58f8758",
          "vrf": "0ada6c25d62db5e1e35d3df727635afa943b9e8a123ab83785e2281605b09ce2",
        },
        "b00470cd193d67aac47c373602fccd4195aad3002c169b5570de1126": Object {
          "delegate": "b3b539e9e7ed1b32fbf778bf2ebf0a6b9f980eac90ac86623d11881a",
          "vrf": "0ff0ce9b820376e51c03b27877cd08f8ba40318f1a9f85a3db0b60dd03f71a7a",
        },
        "b260ffdb6eba541fcf18601923457307647dce807851b9d19da133ab": Object {
          "delegate": "7c64eb868b4ef566391a321c85323f41d2b95480d7ce56ad2abcb022",
          "vrf": "7fb22abd39d550c9a022ec8104648a26240a9ff9c88b8b89a6e20d393c03098e",
        },
        "ced1599fd821a39593e00592e5292bdc1437ae0f7af388ef5257344a": Object {
          "delegate": "de7ca985023cf892f4de7f5f1d0a7181668884752d9ebb9e96c95059",
          "vrf": "c301b7fc4d1b57fb60841bcec5e3d2db89602e5285801e522fce3790987b1124",
        },
        "dd2a7d71a05bed11db61555ba4c658cb1ce06c8024193d064f2a66ae": Object {
          "delegate": "1e113c218899ee7807f4028071d0e108fc790dade9fd1a0d0b0701ee",
          "vrf": "faf2702aa4893c877c622ab22dfeaf1d0c8aab98b837fe2bf667314f0d043822",
        },
        "f3b9e74f7d0f24d2314ea5dfbca94b65b2059d1ff94d97436b82d5b4": Object {
          "delegate": "fd637b08cc379ef7b99c83b416458fcda8a01a606041779331008fb9",
          "vrf": "37f2ea7c843a688159ddc2c38a2f997ab465150164a9136dca69564714b73268",
        },
      },
      "initialFunds": Object {},
      "maxKESEvolutions": 62,
      "maxLovelaceSupply": "45000000000000000",
      "networkId": "Testnet",
      "networkMagic": 1,
      "protocolParams": Object {
        "a0": 0.3,
        "decentralisationParam": 1,
        "eMax": 18,
        "extraEntropy": Object {
          "tag": "NeutralNonce",
        },
        "keyDeposit": 2000000,
        "maxBlockBodySize": 65536,
        "maxBlockHeaderSize": 1100,
        "maxTxSize": 16384,
        "minFeeA": 44,
        "minFeeB": 155381,
        "minPoolCost": 340000000,
        "minUTxOValue": 1000000,
        "nOpt": 150,
        "poolDeposit": 500000000,
        "protocolVersion": Object {
          "major": 2,
          "minor": 0,
        },
        "rho": 0.003,
        "tau": 0.2,
      },
      "securityParam": 2160,
      "slotLength": 1,
      "slotsPerKESPeriod": 129600,
      "staking": Object {
        "pools": Object {},
        "stake": Object {},
      },
      "systemStart": "2022-06-01T00:00:00Z",
      "updateQuorum": 5,
    },
  },
}
`;
